### ISSUE TITLE
BUG : unhandled promise rejection errors on android when calling `rudderClient.setup`

### ISSUE THREAD

**Issue by sam1463** [rudderlabs/rudder-sdk-react-native#173]
**Describe the bug**
Please provide the following information:

1. I'm importing `rudderClient` and calling `setup` on app launch, and I get a bunch of errors:
My code:
```
import rudderClient from '@rudderstack/rudder-sdk-react-native';

await rudderClient.setup(CONSTANTS.RUDDERSTACK.WRITE_KEY, {
    dataPlaneUrl: CONSTANTS.RUDDERSTACK.DATA_PLANE_URL,
    trackAppLifecycleEvents: true,
    recordScreenViews: true,
});
```
Errors:
```
 ERROR  Unhandled promise rejection [RangeError: Maximum call stack size exceeded.]
 ERROR  Unhandled promise rejection [RangeError: Maximum call stack size exceeded.]
 ERROR  Unhandled promise rejection [RangeError: Maximum call stack size exceeded.]
 ERROR  Unhandled promise rejection [RangeError: Maximum call stack size exceeded.]
 ERROR  Unhandled promise rejection [RangeError: Maximum call stack size exceeded.]
 ERROR  Unhandled promise rejection [RangeError: Maximum call stack size exceeded.]
 ERROR  Unhandled promise rejection [RangeError: Maximum call stack size exceeded.]
 ERROR  Unhandled promise rejection [TypeError: Object is not a constructor]
```

If I just don't call this method, and don't use rudderClient anywhere, the errors don't appear.

This is in a fairly empty new project which is using expo via custom dev client (not Expo Go). This error doesn't show up on iOS. Also I do have `mavenCentral()` in the appropriate spots in the project-level `build.gradle`.

**Which platform is the issue occurring on**
- [X] android

**To Reproduce**
1. Initialise React-Native SDK in an android app
2. Call `rudderClient.setup`
3. See the error warnings in console + red box at the bottom of android emulator.

**Expected behavior**

**Screenshots**

**Version of the _React-Native_ SDK**
Please provide the following information:

1. The version of the Rudder React-Native SDK you are using: 1.6.1

**SDK initialisation snippet**
Share the code snippet used for initializing the React-Native SDK.

**Framework version of the SDK**
react-native: 0.70.5
expo: 47.0.12

**Check for Correct Usage of _writeKey_ and _dataPlaneUrl_**
Confirm that the correct `writeKey` and `dataPlaneUrl` are utilized during SDK initialization. ✅

**Comment by 1abhishekpandey** [comment]
Thank you for bringing this to our attention. We will investigate the issue.

**Comment by 1abhishekpandey** [comment]
Hello, @sam1463,

I attempted to recreate this issue on a sample React-Native app built using `nx` (refer to our [sample app](https://github.com/rudderlabs/rudder-sdk-react-native/tree/master/apps/example)) and the `npx react-native init <PROJECT-NAME>`. I did not observe the issue on either of the sample apps, which suggests that the problem may be specific to `Expo`. We will continue to investigate the issue further.

**Comment by robert-nash** [comment]
In case it is helpful to others, I believe this can also be a problem on iOS in version 1.6.1

I spent a while investigating a bug where my iOS Expo app would crash on first load if @rudderstack/rudder-sdk-react-native was imported but only in production mode. In development mode the SDK would function correctly.

I tracked down the bug to some of the promise polyfill related code which ended up in the compiled index.js file for the SDK. It seems that https://github.com/rudderlabs/rudder-sdk-react-native/pull/175 which aims to fix this issue resolves my crash. This makes sense given in the pull request the compiled index.js file contains none of the thousands of lines of polyfill code it did previously, presumably as a result of changing useBuiltIns from  "usage" to "entry" in the babel config.

I am eagerly awaiting the release of version 1.6.2 to confirm everything is sorted.

**Comment by robert-nash** [comment]
1.6.2 resolved the problem I was having on iOS

**Comment by 1abhishekpandey** [comment]
Glad to know that @robert-nash.

**Comment by 1abhishekpandey** [comment]
@sam1463 Please use the latest version of our [RudderStack React-Native SDK v1.6.2](https://www.npmjs.com/package/@rudderstack/rudder-sdk-react-native) which includes a fix for the unhandled promise rejection issue. Kindly let us know if the issue has been resolved after using this version.

**Comment by sam1463** [comment]
All fixed on 1.6.2. Thanks!
